tao_compiler/mlir/disc/tests/regression/algebra_simplifier.cc

```diff
@@ -0,0 +1,35 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/compiler/mlir/disc/tests/mlir_feature_test.h"
+#include "tensorflow/compiler/mlir/disc/tests/mlir_test.h"
+#include "tensorflow/core/platform/test.h"
+
+namespace mlir_test {
+
+const std::string c_ft_path =
+    "tensorflow/compiler/mlir/disc/tests/regression/data/";
+
+TEST(AlgebraSimplifier, PowF32DynamicTest) {
```

Comment:
Is it necessary to add a regression test for implicit broadcast removal with` DISC_ENABLE_SHAPE_CONSTRAINT_IR=true`?

Reply:
I already add both `DISC_ENABLE_SHAPE_CONSTRAINT_IR=true` and `DISC_ENABLE_SHAPE_CONSTRAINT_IR =false` to all UTs. See code here:
https://github.com/alibaba/BladeDISC/pull/329/files#diff-da2c1053cd526401fd4515ff2f61e79af6cd00a436119ca646ad9c405c4768c8